### DISCUSSION TITLE
Initialize Superset with my source database to query on.

### DISCUSSION THREAD

**Discussion by tadeuif** [apache/superset#11862] [General]
Guys, superset is a amazing application, but i'm "missing" something.
Is there a way of initializing Superset with my DB already connected? When we run at the first time, the examples DB is there ready to work. I'm wondering if i'm missing something to do that.
Can anyone help?
Thanks in advance.

**Comment by srinify** [comment]
Hey @tadeuif ! Are you using Superset through Docker Compose? You'll notice in the config file for Docker Compose (which lives in `docker-compose.yml` that there's a little Postgres database that's spun up (along with some other nifty services!):

https://github.com/apache/superset/blob/master/docker-compose.yml#L40

If you want to add your own database / connection in Superset, you'll have to do that via the Superset UI. Note that you can also add new database _drivers_ as well .. which many databases will need if you want Superset to talk to them:

https://preset.io/blog/2020-05-18-install-db-drivers/#2-install-driver-locally

**Comment by graham2071** [comment]
If someone has the same need, you can use the [POST /api/v1/database/](https://superset.apache.org/docs/api) to create the database connection once superset container is healthy.

Example:

```sh
curl --location --request POST "http://localhost:8088/api/v1/database/" \
--header "X-CSRFToken: $csrf_token" \
--header "Authorization: Bearer $access_token" \
--header "Content-Type: application/json" \
--data "@apache-drill.json"
```

with `apache-drill.json`:

```json
{
  "database_name": "Apache Drill",
  "sqlalchemy_uri": "drill+sadrill://drill:8047/mongo?use_ssl=False",
  "cache_timeout": null,
  "expose_in_sqllab": true,
  "allow_run_async": false,
  "allow_ctas": false,
  "allow_cvas": false,
  "allow_dml": false,
  "allow_csv_upload": false,
  "extra": "{\"allows_virtual_table_explore\": true}",
  "version": "1.0.0"
}
```
